### PR TITLE
Catch panics and return 500 errors

### PR DESCRIPTION
**Pull request by asalkeld**
part of https://github.com/nitrictech/cli/issues/290

### CODE REVIEW

File: faas/loop.go [hunk 1]
```diff
@@ -23,23 +23,56 @@ import (
...
+
+				err = faasSendResponse(stream, ctx, srvrMsg.GetId())
+				if err != nil {
+					fmt.Println("Error sending error response", err)
+				}
+
+				errorCh <- fmt.Errorf("%v", rErr)
+			}
+		}
+	}()
```
Comment (tjholm) [inline faas/loop.go:62]: The scope of this seems a bit off, we want to recover in the case where the panic occurs in the scope of handler so this should probably be scoped to the for loop so the stream may continue handling requests and the user application continues to run.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: faas/loop.go (hunk 1 above)
Comment (asalkeld) [reply faas/loop.go:62]: mmm, I am not sure that handling panics and continuing is a good idea..

Comment (asalkeld) [reply faas/loop.go]: OK, much improved now :-)

**Review by davemooreuws — APPROVED** [review]
LGTM

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**